Regression_test/CMakeLists.txt

```diff
@@ -23,3 +23,4 @@ add_definitions("-DSOFA_SRC_DIR=\"${CMAKE_SOURCE_DIR}\"")
 target_link_libraries(${PROJECT_NAME} SofaCore SofaGTestMain SofaComponentBase SofaComponentCommon SofaComponentGeneral SofaComponentAdvanced SofaComponentMisc)
 target_link_libraries(${PROJECT_NAME} SofaExporter SofaValidation)
 
+target_compile_features(${PROJECT_NAME} PRIVATE cxx_std_17)
```

Comment:
Is this multiplatform?
In SOFA we do
```cmake
set(CMAKE_CXX_STANDARD 11)
set(CMAKE_CXX_STANDARD_REQUIRED ON)
```
What is the difference?

Reply:
```
set(CMAKE_CXX_STANDARD 17)
set(CMAKE_CXX_STANDARD_REQUIRED ON)
```
is only used for the current projects being build. Linking against those libraries won't set the good standard.

If target A has headers with c++17 features, and I link my external library against it, using the set(...) in the project A won't tell my library that it need c++17 to use those headers.

I really think we should be using `target_compile_features` instead of relying on global variables such as CMAKE_CXX_STANDARD